integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkStreamingTest.java

```diff
@@ -296,10 +308,46 @@ void testKafkaSourceToBatchSink() throws TimeoutException, StreamingQueryExcepti
           assertEquals("file", event.getOutputs().get(0).getNamespace());
         });
 
+    // make sure events share the same runUuid and jobName
+    assertThat(
+            events.stream()
+                .filter(
+                    e ->
+                        e.getRun().getFacets() == null
+                            || e.getRun().getFacets().getParent()
+                                != null) // ignore application events
+                .map(x -> Pair.of(x.getRun().getRunId(), x.getJob().getName()))
+                .distinct()
+                .collect(Collectors.toList()))
+        .hasSize(1);
+
+    List<String> eventTypes =
```

Comment:
One thing that this PR is not solving is that, events generated for the in-the-middle SQL queries should not contain job and run facets, at least the processing type facet. To be fixed in case a community has a consensus on which approach to solve.

Reply:
@pawel-big-lebowski i want to pick up this PR or at least get a running version for our needs.

i know that you have already closed this PR, but i wanted to ask if there has any consensus been reached to this or where the discussion is made.